resources/compute_virtual_machines.go

```diff
@@ -0,0 +1,608 @@
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("VirtualMachineProperties.DiagnosticsProfile.BootDiagnostics.StorageURI"),
+			},
+			{
+				Name:        "availability_set_id",
+				Description: "Resource Id",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("VirtualMachineProperties.AvailabilitySet.ID"),
+			},
+			{
+				Name:        "virtual_machine_scale_set_id",
+				Description: "Resource Id",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("VirtualMachineProperties.VirtualMachineScaleSet.ID"),
+			},
+			{
+				Name:        "proximity_placement_group_id",
+				Description: "Resource Id",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("VirtualMachineProperties.ProximityPlacementGroup.ID"),
+			},
+			{
+				Name:        "priority",
+				Description: "Specifies the priority for the virtual machine <br><br>Minimum api-version: 2019-03-01 Possible values include: 'Regular', 'Low', 'Spot'",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("VirtualMachineProperties.Priority"),
+			},
+			{
+				Name:        "eviction_policy",
+				Description: "Specifies the eviction policy for the Azure Spot virtual machine and Azure Spot scale set <br><br>For Azure Spot virtual machines, both 'Deallocate' and 'Delete' are supported and the minimum api-version is 2019-03-01 <br><br>For Azure Spot scale sets, both 'Deallocate' and 'Delete' are supported and the minimum api-version is 2017-10-30-preview Possible values include: 'Deallocate', 'Delete'",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("VirtualMachineProperties.EvictionPolicy"),
+			},
+			{
+				Name:        "billing_profile_max_price",
+				Description: "Specifies the maximum price you are willing to pay for a Azure Spot VM/VMSS This price is in US Dollars <br><br> This price will be compared with the current Azure Spot price for the VM size Also, the prices are compared at the time of create/update of Azure Spot VM/VMSS and the operation will only succeed if  the maxPrice is greater than the current Azure Spot price <br><br> The maxPrice will also be used for evicting a Azure Spot VM/VMSS if the current Azure Spot price goes beyond the maxPrice after creation of VM/VMSS <br><br> Possible values are: <br><br> - Any decimal value greater than zero Example: 001538 <br><br> -1 – indicates default price to be up-to on-demand <br><br> You can set the maxPrice to -1 to indicate that the Azure Spot VM/VMSS should not be evicted for price reasons Also, the default max price is -1 if it is not provided by you <br><br>Minimum api-version: 2019-03-01",
+				Type:        schema.TypeFloat,
+				Resolver:    schema.PathResolver("VirtualMachineProperties.BillingProfile.MaxPrice"),
+			},
+			{
+				Name:        "host_id",
+				Description: "Resource Id",
```

Comment:
same as above , we have multiple fields here with the description "resource id" we should try to be more specific